Automattic/wp-calypso
# Add Migration Trial case to plans navigation

Context https://github.com/Automattic/wp-calypso/pull/73504.

Fixes https://github.com/Automattic/wp-calypso/issues/79529

## Proposed Changes

* Change tab name (see context)
* ~~Add a generic isTrialPlan function~~

## Testing Instructions



* Visit `http://calypso.localhost:3000/plans/my-plan/__SLUG__.wpcomstaging.com`
* Check that the two tabs are named "Free trial" and "Plans"

## Pre-merge Checklist
